.github/workflows/test.yaml

```diff
@@ -17,6 +17,9 @@ jobs:
     if: github.event.pull_request.draft == false
     name: Unit
     runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
```

Comment:
I think the lint and the test process themselves will tell you where the issues is and the author should have an idea what changes they made that touched reported part of the code.
Regardless if this becomes an issue this can be reverted.